go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -291,6 +368,45 @@ func newSharedGatewayOpenFlowManager(patchPort, macAddress, gwBridge, gwIntf str
 	dftFlows = append(dftFlows,
 		fmt.Sprintf("cookie=%s, priority=0, table=1, actions=output:NORMAL", defaultOpenFlowCookie))
 
+	// table 2, dispatch from Host -> OVN
+	dftFlows = append(dftFlows,
+		fmt.Sprintf("cookie=%s, table=2, "+
+			"actions=mod_dl_dst=%s,output:%s", defaultOpenFlowCookie, macAddress, ofportPatch))
+
+	// table 3, dispatch from OVN -> Host
+	dftFlows = append(dftFlows,
+		fmt.Sprintf("cookie=%s, table=3, "+
+			"actions=move:NXM_OF_ETH_DST[]->NXM_OF_ETH_SRC[],mod_dl_dst=%s,output:LOCAL",
+			defaultOpenFlowCookie, macAddress))
+
+	// table 4, hairpinned pkts that need to go from OVN -> Host
+	// We need to SNAT and masquerade OVN GR IP, send to table 3 for dispatch to Host
+	if config.IPv4Mode {
+		dftFlows = append(dftFlows,
+			fmt.Sprintf("cookie=%s, table=4,ip,"+
+				"actions=ct(commit,zone=%d,nat(src=%s),table=3)",
+				defaultOpenFlowCookie, OVNMasqCTZone, types.V4OVNMasqueradeIP))
+	}
+	if config.IPv6Mode {
+		dftFlows = append(dftFlows,
+			fmt.Sprintf("cookie=%s, table=4,ip6, "+
+				"actions=ct(commit,zone=%d,nat(src=%s),table=3)",
+				defaultOpenFlowCookie, OVNMasqCTZone, types.V4OVNMasqueradeIP))
```

Comment:
L394: types.V6OVNMasqueradeIP?

Reply:
ah thank you!

Reply:
also remind me to make a doc file to show how the packet flow will work to include in the final PR